pallets/schemas/src/migrations.rs

```diff
@@ -0,0 +1,33 @@
+//! Migrations for the schemas pallet.
+//! Following migrations are required:
+//! - Schema struct from v1 to v2
+//! - #TODO: Post migration cleanup
+//! - Remove the Schema struct and respective storage once migration is complete
+//! - Rename SchemaV2 to Schema and respective storage once migration is complete
+use super::*;
+use crate::types::{Schema, SchemaV2};
+use codec::{Decode, Encode};
+use common_primitives::schema::{SchemaId, SchemaSettings};
+use frame_support::pallet_prelude::Weight;
+
+pub fn migrate_schema_to_schema_v2<T: Config>() -> Weight {
+	let mut weight: Weight = Weight::zero();
+	let schema_count = <Schemas<T>>::iter().count();
```

Comment:
Maybe more efficient to use `iter().enumerate()` here instead of `count()`? Would save having to re-fetch the schema later on.